test/tests.py

```diff
@@ -305,3 +310,34 @@ def test_0006_OSB(self):
             if source_count != target_count:
                 self.assertEqual(source_count, target_count, f'{index}: doc counts do not match - '
                                                              f'Source = {source_count}, Target = {target_count}')
+
+    def test_0007_timeBetweenRequestsOnSameConnection(self):
+        # This test will verify that the replayer functions correctly when
+        # requests on the same connection on the proxy that has a minute gap
+        seconds_between_requests = 60  # 1 minute
+
+        # Set a consistent session to be used for this test
+        session = Session()
+        adapter = HTTPAdapter(pool_connections=1, pool_maxsize=1, max_retries=1)
+        session.mount('http://', adapter)
+        session.mount('https://', adapter)
+
+        index_name = f"test_0007_{self.unique_id}"
+
+        number_of_docs = 3
+
+        for doc_id_int in range(number_of_docs):
+            doc_id = str(doc_id_int)
+            proxy_response = create_document(self.proxy_endpoint, index_name, doc_id, self.source_auth,
+                                             self.source_verify_ssl, session)
+            self.assertEqual(proxy_response.status_code, HTTPStatus.CREATED)
+
+            if doc_id_int + 1 < number_of_docs:
+                time.sleep(seconds_between_requests)
+
+        try:
+            for doc_id_int in range(number_of_docs):
```

Comment:
Do you need a sleep before this, or some other way to poll on the target cluster in case the document hasn't been replayed yet?

Reply:
the assert_source_target_doc_match does a poll on the target. No sleep is needed

Reply:
where is the poll?  I didn't see it.

Reply:
>        target_response = retry_request(get_document, args=(self.target_endpoint, index_name, doc_id,
>                                                           self.target_auth, self.target_verify_ssl),
>                                        expected_status_code=HTTPStatus.OK)
Its the retry_request. That's how the existing tests check, it performs a backoff and retry until it get's the desired status code or preconfigured max number or retries occur